include/soci/connection-parameters.h

```diff
@@ -32,26 +32,33 @@ class SOCI_DECL connection_parameters
 
     // Retrieve the backend and the connection strings specified in the ctor.
     backend_factory const * get_factory() const { return factory_; }
-    void set_connect_string(const std::string & connectString) { connectString_ = connectString; }
+    void set_connect_string(std::string const & connectString)
+    {
+        connectString_ = connectString;
+        parse();
+    }
     std::string const & get_connect_string() const { return connectString_; }
 
     // Set the value of the given option, overwriting any previous value.
-    void set_option(const char * name, std::string const & value)
+    void set_option(std::string const & name, std::string const & value)
```

Comment:
To be more precise, here is my understanding of the case like this
```
    string key{"k"}, value{"v"};
    map<string, string> m;
    m[key] = value;
```

AFAIU, key passed by `string const&` is still eventually emplaced in `m` by `std::forward<string>(key)` which in fact makes a copy. The `value` is copied too, obviously.

In this particular case, passing key by `char*` or `string&` makes no difference to me, both are copied. Please, correct me if I'm wrong.